WhiteDentalClinic/WhiteDentalClinic.Api/Controllers/AppointmentController.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System.ComponentModel.DataAnnotations;
+using WhiteDentalClinic.Application.Models;
+using WhiteDentalClinic.Application.Models.AppointmentModel;
+using WhiteDentalClinic.Application.Services.Interfaces;
+
+namespace WhiteDentalClinic.Api.Controllers
+{
+    public class AppointmentController : ApiController
+    {
+        readonly IAppointmentService _appointmentService;
+        public AppointmentController(IAppointmentService appointmentService)
+        {
+            _appointmentService = appointmentService;
+        }
+
+        [HttpGet("allapp/customer{customerRequestId}")]
+        public ActionResult<IEnumerable<AppointmentResponseModel>> GetAllAppointmentsByCustomer(Guid customerRequestId)
+        {
+            try
+            {
+                return Ok(ApiGenericsResult<IEnumerable<AppointmentResponseModel>>.Success(_appointmentService.GetAllAppointmentsByCustomer(customerRequestId)));
+            }
+            catch (Exception ex)
+            {
+                if(_appointmentService.GetAllAppointments().ToList().First(x => x.CustomerId == customerRequestId) ==null)
+                {
+                    return NotFound(ApiGenericsResult<AppointmentResponseModel>.Failure(new[] { $"{ex.Message}" }));
+                }
+                return BadRequest(ApiGenericsResult<AppointmentResponseModel>.Failure(new[] { $"{ex.Message}" }));
+            }
+        }
+
+        [HttpGet("allapp/dentist{dentistRequestId}")]
+        public ActionResult<IEnumerable<AppointmentResponseModel>> GetAllAppointmentsByDentist(Guid dentistRequestId)
+        {
+            try
+            {
+                return Ok(ApiGenericsResult<IEnumerable<AppointmentResponseModel>>.Success(_appointmentService.GetAllAppointmentsByDentist(dentistRequestId)));
+            }
+            catch (Exception ex)
+            {
+                if (_appointmentService.GetAllAppointments().ToList().Select(x => x.DentistId == dentistRequestId) ==null)
+                {
+                    return NotFound(ApiGenericsResult<AppointmentResponseModel>.Failure(new[] { $"{ex.Message}" }));
+                }
+                return BadRequest(ApiGenericsResult<AppointmentResponseModel>.Failure(new[] { $"{ex.Message}" }));
+            }
+        }
+
+
+        [HttpGet("{id:guid}")]
```

Comment:
"api/appointments" from BaseController (after pluralizing controller name) so "{appointmentId}" would be enough here.